File: src/main/java/kr/co/bookand/backend/config/security/SecurityUtils.java
package kr.co.bookand.backend.config.security;

import kr.co.bookand.backend.account.domain.Account;
import kr.co.bookand.backend.account.exception.AccountException;
import kr.co.bookand.backend.account.repository.AccountRepository;
import kr.co.bookand.backend.common.exception.ErrorCode;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;

@RequiredArgsConstructor
public class SecurityUtils {

    private final AccountRepository accountRepository;
    public static String getCurrentAccountEmail() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        if (authentication == null || authentication.getName() ==null){
            throw new AccountException(ErrorCode.NOT_FOUND_MEMBER, null);
        }
        return authentication.getName();
    }

    public static Account getCurrentAccount(AccountRepository accountRepository) {
        String currentAccountEmail = getCurrentAccountEmail();
        return accountRepository.findByEmailAndVisibilityTrue(currentAccountEmail).orElseThrow(() -> new AccountException(ErrorCode.NOT_FOUND_MEMBER, null));
    }

}

File: src/main/kotlin/kr/co/bookand/backend/config/security/SecurityUtils.kt
package kr.co.bookand.backend.config.security

import kr.co.bookand.backend.account.model.Account
import kr.co.bookand.backend.account.repository.AccountRepository
import kr.co.bookand.backend.common.ErrorCode
import kr.co.bookand.backend.common.exception.BookandException
import org.springframework.security.core.context.SecurityContextHolder

object SecurityUtils {
    fun getCurrentAccountEmail(): String =
        SecurityContextHolder.getContext().authentication?.name
            ?: throw BookandException(ErrorCode.NOT_FOUND_MEMBER)

    fun getCurrentAccount(accountRepository: AccountRepository): Account =
        accountRepository.findByEmail(getCurrentAccountEmail())
            ?: throw BookandException(ErrorCode.NOT_FOUND_MEMBER)
}
